src/spec-common/injectHeadless.ts

```diff
@@ -484,38 +484,49 @@ async function runLifecycleCommand({ lifecycleHook }: ResolverParameters, contai
 			},
 			onDidChangeDimensions: lifecycleHook.output.onDidChangeDimensions,
 		}, LogLevel.Info);
-		try {
-			const remoteCwd = containerProperties.remoteWorkspaceFolder || containerProperties.homeFolder;
-			async function runSingleCommand(postCommand: string | string[], name?: string) {
-				const progressDetail = typeof postCommand === 'string' ? postCommand : postCommand.join(' ');
-				infoOutput.event({
-					type: 'progress',
-					name: progressName,
-					status: 'running',
-					stepDetail: progressDetail
-				});
+		const remoteCwd = containerProperties.remoteWorkspaceFolder || containerProperties.homeFolder;
+		async function runSingleCommand(postCommand: string | string[], name?: string) {
+			const progressDetail = typeof postCommand === 'string' ? postCommand : postCommand.join(' ');
+			infoOutput.event({
```

Comment:
Move this event to right before `await Promise.all(commands);`.

Reply:
done. moved this before `await Promise.all(commands);`

Reply:
Actually, since `progressDetail` includes the command to run, maybe this should stay here.

@samruddhikhandale Could you advise on this? I think Codespaces is using the progress events in the UI?

@prathameshzarkar9 I don't see your latest changes on GitHub, did you push the commits?

Reply:
Hi @chrmarti ,
changes suggested by you are incorporated and the new changes also checked-in here in the latest commits.
PR is now updated can you please review it once @samruddhikhandale ?

Reply:
> Actually, since progressDetail includes the command to run, maybe this should stay here.
> @samruddhikhandale Could you advise on this? I think Codespaces is using the progress events in the UI?

@chrmarti  Good catch here! 
Yes, Codespaces is still using `progressDetail` so let's keep it as is.

Reply:
@samruddhikhandale , @chrmarti 
I kept the progressDetails as it is. can you review it once?